### ISSUE TITLE
When using Table with IQueryable and QueryModel parameter an Exception is thrown with Arg_ParamName_Name

### ISSUE THREAD

**Issue by syntax-art** [ant-design-blazor/ant-design-blazor#2154]
Hi,

I'm wondering if anyone else has had the same issue when using Table in Ant Blazor.

When you define the Table and use the method OnChange you get a QueryModel parameter which you can then pass to the ExecuteTableQuery() method.  Inside this method, the Expression functions are called to build the query which would help filter the IQueryable.  When you do a simple filter using the table UI such as Contains(), the OnChange method is executed and you can pass the QueryModel parameter to ExecuteTableQuery() which then throws an Exception:

`System.ArgumentNullException: 'Value cannot be null. Arg_ParamName_Name'`

An example of the queryable is here

```
_total = queryable.Count();

IEnumerable<Product>_data = queryable.ExecuteTableQuery(query).Skip(_pageIndex).Take(_pageSize).ToList();
```

This is not a custom filter, but using the built-in filters on the table.  The data type of the bound column is String.

The table component

```
<Table DataSource="_data"
       TItem="Product"
       @bind-PageIndex="_pageIndex"
       @bind-PageSize="_pageSize"
       Total="_total"
       OnChange="OnChange">
    <Column @bind-Field="context.Title"
            TData="string"
            Sortable
            Filterable />
</Table>
```

Hope someone can help?

Thank you

### Further technical details
- AntDesign Nuget Package version 0.10.2
- The IDE VS 2022 Using Blazor Server

**Comment by anranruye** [comment]
Can you post a repro?

I can not reproduce this issue by the following code:
```
<Table DataSource="_data" OnChange="OnChange" TItem="Data" PageIndex="_pageIndex"
       PageSize="_pageSize" Total="_total" RemoteDataSource>
    <Column TData="Guid"
            @bind-Field="context.Id"
            Sortable
            Filterable />

    <Column TData="string"
            @bind-Field="context.Name"
            SorterCompare="@((a,b)=> string.Compare(a,b))"
            SortDirections="new[] { SortDirection.Descending }"
            Filterable />

    <Column TData="string"
            DataIndex="Address"
            SorterCompare="@((a,b)=> string.Compare(a,b))"
            SortDirections="new[] { SortDirection.Descending, SortDirection.Ascending }"
            Filterable />

    <Column TData="DateTime"
            @bind-Field="context.BirthDate"
            Format="yyyy/MM/dd"
            Sortable
            Filterable />

    <Column TData="bool"
            @bind-Field="context.IsActive"
            Sortable
            Filterable />

    <Column TData="Gender"
            @bind-Field="context.Gender"
            Sortable
            Filterable />

    <Column TData="Hobbies?"
            @bind-Field="context.Hobbies"
            Sortable
            Filterable />
</Table>

@using AntDesign.TableModels;
@using System;
@code {

    class Data
    {
        public Data(string name, string address, DateTime birthDate, bool isActive, Gender gender, Hobbies? hobbies)
        {
            Id = Guid.NewGuid();
            Name = name;
            Address = address;
            BirthDate = birthDate;
            IsActive = isActive;
            Gender = gender;
            Hobbies = hobbies;
        }

        public Data()
        {
            Id = Guid.NewGuid();
        }

        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Address { get; set; }
        public DateTime BirthDate { get; set; }
        public bool IsActive { get; set; }
        public Gender Gender { get; set; }
        public Hobbies? Hobbies { get; set; }
    }

    enum Gender
    {
        Male,
        Female
    }

    [Flags]
    enum Hobbies
    {
        Reading = 1,
        Sport = 2,
        Cooking = 4
    }

    Data[] _allData;
    int _pageIndex = 1;
    int _pageSize = 10;
    int _total = 100;

    Data[] _data;

    protected override void OnInitialized()
    {
        _allData = new Data[100];
        for (int i = 0; i < 100; i++)
        {
            var data = new Data();
            data.Name = (i % 3) switch
            {
                1 => $"John Brown{i}",
                2 => $"Sara Green{i}",
                0 => $"Joe Black{i}",
                _ => ""
            };
            data.Address = (i % 3) switch
            {
                1 => $"New York{i}",
                2 => $"London{i}",
                0 => $"Sidney{i}",
                _ => ""
            };
            data.BirthDate = new DateTime(1970 + i, 1, 1);
            data.Gender = i % 2 == 0 ? Gender.Female : Gender.Male;
            data.Hobbies = (Hobbies)(i % 7) + 1;
            data.IsActive = i % 2 == 0;
            _allData[i] = data;
        }
    }

    void OnChange(QueryModel<Data> query)
    {
        _pageIndex = query.PageIndex;
        _pageSize = query.PageSize;
        var queryable = _allData.AsQueryable().ExecuteTableQuery(query);
        _total = queryable.Count();
        _data = queryable.CurrentPagedRecords(query).ToArray();
    }
}
```

**Comment by ryan-haworth** [comment]
Hi,

Thank you for your response

I can see that you've built your collection in memory, but I'm using MongoDB.  You are able to get an IQueryable from their API and I'm able to get your extension methods.   I've tried adding some of the missing parts from your solution but it's still not working.  It will always throw an exception after you have created your first filter.  The IQueryable will work the first time you load as the expression has not been created the first time you get the data.

I've used the IQueryable with other controls in the past so I know MongoDB API is capable of filtering the query later.  Is it possible that the expression isn't correct?  

Have you tried this with EntityFramework or any other ORM Framework?

I am able to create a solution that can build a collection in MongoDB and then perform a bind for the data if you would like to try this with MongoDB?

Thank you

**Comment by syntax-art** [comment]
Hi,

Thank you for your solution. The problem was in the V2 Linq Driver for MongoDB - AntBlazor Table works as expected (my apologies)

I'm currently using MongoDB so I've added some changes from your solution above and also found that I needed to make some more changes to the Mongo Installation to make this work as there is a bug in their V2 LINQ driver.

This solution works
Upgrade to MongoDB Community Server 5.0
Install C# Driver 2.14.1

Set the Linq provider to V3 (This fixes the bug in [Dynamic Linq Expression Fix](https://jira.mongodb.org/browse/CSHARP-2422)

```
var _clientSettings = MongoClientSettings.FromConnectionString("mongodb://127.0.0.1:27017/");
_clientSettings.LinqProvider = LinqProvider.V3;
_client = new MongoClient(_clientSettings);
```

Thank you for your help, I posted this solution for anyone who may get this issue in future.
